ptest/cases/autonomous_mission_manager_case.py

```diff
@@ -0,0 +1,129 @@
+            return False
+        return True
+    
+    def continue_mission(self):
+        #check for operating leader state
+        leader_state_functional = self.state_check(self.leader, "Leader")
+        follower_state_functional = self.state_check(self.follower, "Follower")
+        if not leader_state_functional:
+            if follower_state_functional:
+                self.follower.write_state("pan.state", Enums.mission_states["standby"])
+            elif self.follower.read_state("pan.state") != str(Enums.mission_states["standby"]):
+                self.follower.write_state("pan_state", Enums.mission_states["safehold"])
+            return False
+
+        #check faulting state for follower state
+        if not follower_state_functional:
+            self.leader.write_state("pan.state", Enums.mission_states["standby"])
+            return False
+
+        #check time since last comms
+        leader_time_since_comms = time.time() - self.leader_time_last_comms 
+        if(leader_time_since_comms > self.comms_time_threshold):
+            self.logger.put("Leader is experiencing comms blackout. Ending mission." + str(leader_time_since_comms))
+            self.leader.write_state("pan.state", Enums.mission_states["standby"])
+            self.follower.write_state("pan.state", Enums.mission_states["standby"])
+            return False
+        follower_time_since_comms = time.time() - self.follower_time_last_comms 
+        if(follower_time_since_comms > self.comms_time_threshold):
+            self.logger.put("Follower is experiencing comms blackout. Ending mission." + str(follower_time_since_comms))
+            self.leader.write_state("pan.state", Enums.mission_states["standby"])
+            self.follower.write_state("pan.state", Enums.mission_states["standby"])
+            return False
+
+        return True
+    
+    def propagate_orbits(self, vals):
+        return vals
+
+
+    @property
+    def sim_configs(self):
+        configs = ["truth/ci", "truth/base"]
+        configs += ["sensors/base"]
+        return configs
+
+    @property
+    def sim_model(self):
+        return DualAttitudeOrbitGnc
+
+    @property
+    def sim_mapping(self):
+        return "ci_mapping.json"
+        
+    @property
+    def sim_duration(self):
+        return float("inf")
+
+    @property
+    def initial_state_leader(self):
+        return "leader"
+    @property
+    def initial_state_follower(self):
+        return "follower"
+
+    @property
+    def fast_boot_leader(self):
+        return True
+    @property
+    def fast_boot_follower(self):
+        return True
+        
+
+    def run_case_fullmission(self):
+
+        #setup
+        self.using_radios = self.radio_follower != None #assumes only using one physical radio
+        self.leader = self.flight_controller_leader
+        self.follower = self.radio_follower if self.using_radios else self.flight_controller_follower
```

Comment:
```suggestion
        self.leader = self.radio_leader if self.radio_leader !=None else self.flight_controller_leader
        self.follower = self.radio_follower if self.using_radios else self.flight_controller_follower
```
The `using_radios` variable isn't used anywhere else in the code as far as I can tell, so you could a) get rid of it, b) increase parity between leader and follower initializations.

Reply:
Also are you guys making the AMC satellite-free? In that case you wouldn't need the radio logic right

Reply:
I think this wouldn't work because both radios are initialized, but we only have one physical radio to test on which is why the leader is set to the flight_controller automatically. We will eventually have no satellites, but we will still need the radio_session connection to read and send uplinks.

Reply:
Sure, ok. So it's eventually on the docket though to make the AMC satellite-free